### ISSUE TITLE
Prettier sometimes failes to properly format Angular's control flow statements

### ISSUE THREAD

**Issue by diegovilar** [prettier/prettier#16281]
Strangely, this does not happen often, but I've noticed that Prettier doesnt seem to properly format the following Angular template. It brings the `@if` clauses together and doesn't ident the html content inside said `@if` clauses:

**Prettier 3.0.2**
[Playground link](https://prettier.io/playground/?spm=a2c6h.13046898.publish-article.39.1c4e6ffaNVrRIq#N4Igxg9gdgLgprEAuEABAlgMwAQAoCOArnAE4CeAoiSRCbgJQDc2AhgM7ak0n3bAA6UbNgA86EiwC0XWpJYAbUjGwBtcSyrcAugF5+IGSX3YA9AD5BAX0GCMOXAAdF6GBAbN22APpO4LiLzAwoLCYhKSDixQcPLY6gAqZA5weiAwElBskSQIMPoWQsKiyZKQORHOrpIw6AC2MejRqr7+uvo+lRDG5iGiJuoRUTEFvSIlACYs4xBsFX5VkdGxKi2ubSAd810geNAAwgAWUQDmcPSp0AAKnYcncLgA5ArwEmAsEA-03SOFY3ByNQAbiw3hA5v5Bktmp11pt-MZcPsjlBTud9FcbsjTo8WECQe9Pt8bL8SphGgp0AAvFgkcELIbLVYQWFMhFIu5okAYra3FH3B5kqAU6mgwk7HpQUYlOAADzA8kI4jpYMWMWhWxZnTZUF5qIuUGuPKx-Nl8sIbAJX3FBUsIAANFyHDVoGxkKAaTQAO6XGkIV0oBSelhkV0OgBGrwA1nAYABlFj1AAyjTgyEwCjYcHDUZjsciYEax2Q6WIDsztXQxZIpZAbELigAioQIPA0xmsyAAFZsGWx+twJst1NIdPyTMOohDy40Bz+kDsSTRODjZf2tISdDyQt7CC1WosZDz+TyNd1lGKACCMHS6DDhHgl1IyeibbHHYOMFq8gA6gcXHAshBOBYz9Fx0EBFwyEPMA2FDEBAWIABJKAV1gWMwBIdAnQvFDYxgMhFFfccQAcGhM2-CQHEPUiANIQFUwdRpMxIGBpxYY59yIjtsmYw8TkIeQaTXUjGhgb90HGGADmQABGAAWAAGB0ciIcQ4DYjiDxHdsHRgFgw3EyTpKQAAmB1zTgeJ9P9UdiLgWow2XFdxkTfj2LgAAxWh92vQs+PvCAQEsSwgA)

```sh
--parser angular
--print-width 140
```

**Input:**

```html
@if (queryError(); as error) {
  <ira-error-alert [iraError]="error" />
}

@if (pleito(); as _pleito) {  
  <ira-panel iraType="transparent">
    <pe-core-pleito-timeline [pleito]="_pleito" />
  </ira-panel>

  <pe-dados-pleito-panel [pleito]="_pleito" (onChange)="onPleitoChange('alteracao')" />

  <pe-ativacao-pleito-panel [pleito]="_pleito" (onChange)="onPleitoChange('ativacao')" />

  <pe-finalizar-pleito-panel [pleito]="_pleito" (onChange)="onPleitoChange('finalizacao')" />

  <pe-excluir-pleito-panel [pleito]="_pleito" (onChange)="onPleitoChange('exclusao')" />
}
```

**Output:**

```html
@if (queryError(); as error) {
<ira-error-alert [iraError]="error" />
} @if (pleito(); as _pleito) {
<ira-panel iraType="transparent">
  <pe-core-pleito-timeline [pleito]="_pleito" />
</ira-panel>

<pe-dados-pleito-panel [pleito]="_pleito" (onChange)="onPleitoChange('alteracao')" />

<pe-ativacao-pleito-panel [pleito]="_pleito" (onChange)="onPleitoChange('ativacao')" />

<pe-finalizar-pleito-panel [pleito]="_pleito" (onChange)="onPleitoChange('finalizacao')" />

<pe-excluir-pleito-panel [pleito]="_pleito" (onChange)="onPleitoChange('exclusao')" />
}

```

**Expected behavior:**

**Comment by fisker** [comment]
Can't reproduce in playground link. Maybe you use the wrong parser?

**Comment by diegovilar** [comment]
> Can't reproduce in playground link. Maybe you use the wrong parser?

Interesting, the ticket description was generated by the playground itself, and it shows I had selected the Angular parser...

Anyways, I've just noticed that the Playground IS working correctly today (🤔) when I set the parser to Angular, so I figured prettier in my VSCode is not detecting the HTML file in question as an Angular component an is using the HTML parser, which indeed does not formatted as expected (why should it, right?).

So, I figure Prettier selects the parser according to the file name pattern, right? anything `*.component.html` or `*.directive.html` it guesses to be Angular, but anything else like `*.page.html` or `*.modal.html` it just uses the html parser, is that it?

If so, is there anyway to fine tune that to make it recognize other patterns as a specific parser?
